doc/admin-guide/files/ssl_server_name.config.en.rst

```diff
@@ -0,0 +1,124 @@
+
+verify_client           One of the values :code:`NONE`, :code:`MODERATE`, or :code:`STRICT`.
+                        By default this is :ts:cv:`proxy.config.ssl.client.certification_level`.
+
+client_cert             The client certificate to use for the outbound connection. If this is relative
+                        it is relative to the path in :ts:cv:`proxy.config.ssl.server.cert.path`.
+                        If not set :ts:cv:`proxy.config.ssl.client.cert.filename` is used.
+
+disable_h2              :code:`true` or :code:`false`. If :code:`false` then HTTP/2 is removed from
+                        the valid next protocol list. It is not an error to set this to :code:`false`
+                        for proxy ports on which HTTP/2 is not enabled.
+
+tunnel_route            Destination as an FQDN and port, separated by a colon '``:``'.
+======================= ==============================================================================
+
+Client verification, via ``verify_client``, correponds to setting
+:ts:cv:`proxy.config.ssl.client.certification_level` for this connection as noted below.
+
+:code:`NONE` -- ``0``
+   Do not request a client certificate.
+
+:code:`MODERATE` - ``1``
+   Request a client certificate and do verification if one is provide, but
+   do not enforce verification. A failure is logged but the connection is
+   allowed.
+
+:code:`STRICT` - ``2``
+   Request a client certificate and require one to be provided and verified.
+   A verification failure will cause the connection to fail.
+
+Upstream (server) verification is similar to client verification except there is always an upstream certificate.
+This is equivalent to setting :ts:cv:`proxy.config.ssl.client.verify.server` for this connection.
+
+:code:`NONE` - ``0``
+   Do not verify the upstream server certificate.
+
+:code:`MODERATE` - ``2``
+   Do verification of the upstream certificate but do not enforce.
+
+:code:`STRICT` - ``1``
+   Require successful validation of the upstream certificate to complete the
+   connection.
+
+If ``tunnel_route`` is specified then none of the certificate verification will be done making those
+values irrelevant for that configuration item. This is option is explained in more detail in :ref:`sni-routing`.
+
+Examples
+========
+
+Disable HTTP/2 for ``no-http2.example.com``.
+
+.. code-block:: Lua
+
+   server_config = {{ fqdn="no-http2.example.com", disable_h2=true }}
+
+Require client certificate verification for ``example.com`` and ``yahoo.com``.
+
+.. code-block:: Lua
+
+   server_config = {
+      { fqdn="example.com", verify_client=STRICT },
+      { fqdn="yahoo.com", verify_client=STRICT }
+   }
+
+Disable server certificate verification for ``trusted.example.com`` and require a valid client certificate.
+
+.. code-block:: Lua
+
+   server_config = {
+      {
+         fqdn="trusted.example.com",
+         verify_origin_server=NONE,
+         verify_client=STRICT
+      }
+   }
+
```

Comment:
When does this file get loaded? (eg, on when changed, on traffic_ctl config reload?). Is it worth noting that you may have to configure remap.config for the fqdm and definitely have to have a matching ssl_multicert.config entry?

Reply:
This file can be reloaded with traffic_ctl config reload given that there is a change in the file. Configuring remap.config should not be a requirement since there is a separate data structure for setting up the outbound connection properties. Also having a matching entry in ssl_multicert.config is not a requirement and I don't think it should be since ATS serves default cert in case a matching cert is not found

Reply:
Was thinking that those questions might be answered in the doc (unless they are silly).

Having a default cert means that there is at least some ssl_multicert coordination :)

Reply:
I made some updates about this. I tried to do more emphasis on this file overriding defaults. Can you specify a client cert in `ssl_multicert.config`?

Reply:
no. Client cert is specified either in records.config or in remap.config(since the config is overridable). I think I will get rid of the overridability of that config since it can be specified from ssl_servername now.